### PR TITLE
Week 7

### PR DESCRIPTION
**Pull request by Dev1LtJ**
Created ThemeContext with dark and light theme based on CSS-Variables.

### CODE REVIEW

File: src/App/App.jsx [hunk 1]
```diff
@@ -5,48 +5,104 @@ import classes from './App.module.css';
...
   render() {
-    const { loggedUser, isLogged } = this.state;
-    const contextState = this.state;
+    const {
+      themeContext,
+      themeContext: { theme },
+      userContext,
+      userContext: { isLogged, loggedUser },
+    } = this.state;
+    applyGlobalTheme(theme);
```
Comment (maximsan) [inline src/App/App.jsx:87]: why is it here ? it should be in life cycle methods

File: src/App/App.jsx [hunk 2]
```diff
@@ -5,48 +5,104 @@ import classes from './App.module.css';
...
+      this.updateThemeContext,
     );
   };
 
+  updateThemeContext = () => {
+    const {
+      themeContext: { theme },
+    } = this.state;
+    setGlobalTheme(theme);
+  };
```
Comment (maximsan) [inline src/App/App.jsx:78]: Actually, why do you hold all context's methods in the App ?  Create you custom Provider component and place them there. you don't need them in App, in constructor

File: src/index.css [hunk 3]
```diff
@@ -12,39 +12,132 @@ body {
...
-
   --button-create-color: #1bc47d;
   --button-create-hover-color: #0ca52f;
-
   --button-disabled-color: #ecedf1;
-
+  --button-onScreen-color: #5385ff;
   --validator-color: #d50000;
+  --modal-overlay-color: rgba(0, 0, 0, 0.5);
+  --modal-background-color: #fff;
```
Comment (maximsan) [inline src/index.css:58]: it looks odd. create css variables for the same colors in root and then call it here like`--modal-background-color: var(--white);` follow DRY principle

File: src/pages/Settings.jsx [hunk 4]
```diff
@@ -1,45 +1,88 @@
...
+          >
+            <span className={classes.themeTitle}>Light</span>
+            <img className={classes.themeImg} src={sunLogo} alt='sun-logo' />
+          </div>
+          <div
+            className={darkThemeButtonClass}
+            role='button'
+            tabIndex={0}
+            onKeyDown={this.selectDarkTheme}
+            onClick={this.selectDarkTheme}
```
Comment (maximsan) [inline src/pages/Settings.jsx:73]: create a component, you have the same peace of code twice

File: src/utilities/context-helpers.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
+import { getFromLocalStorage, setToLocalStorage } from './localStorage-helpers';
+
+export const THEME = 'theme';
+export const THEME_LIGHT = 'light';
+export const THEME_DARK = 'dark';
```
Comment (maximsan) [inline src/utilities/context-helpers.js:5]: Why did you not create enum for theme too ?

**Review by maximsan — CHANGES_REQUESTED** [review]
I left some notes. Please review them and try to fix. Good refactoring. Thanks!

File: src/utilities/context-helpers.js (hunk 5 above)
Comment (Dev1LtJ) [reply src/utilities/context-helpers.js:5]: Created:
```suggestion
export const THEMES = {
  THEME: 'theme',
  LIGHT: 'light',
  DARK: 'dark',
};
```

File: src/pages/Settings.jsx (hunk 4 above)
Comment (Dev1LtJ) [reply src/pages/Settings.jsx:73]: Created: 
```suggestion
const ThemeButton = ({ children, onClick, isActive }) => {
  const classList = isActive ? `${classes.item} ${classes.active}` : `${classes.item}`;

  return (
    <button type='button' onClick={onClick} className={classList}>
      <span>{children}</span>
      <img className={classes.img} src={children === THEMES.DARK ? moonLogo : sunLogo} alt='themeButton-logo' />
    </button>
  );
};
```

File: src/index.css (hunk 3 above)
Comment (Dev1LtJ) [reply src/index.css:58]: Created: 
```suggestion
:root {
  --white: #fff;
  --black: #000;
  --backgroundGray: #f6f7fb;
  --lightGray: #ecedf1;
  --darkGray: #7787ab;
  --darkBlue: #0e1935;
  --navyBlue: #122043;
  --lightBlue: #5385ff;
  --skyBlue: #3855a5;
  --red: #d50000;
  --darkRed: #ae0000;
  --wineRed: #500000;
  --crimeRed: #8b0000;
  --lightOrange: #ffa610;
  --orange: #ff7518;
  --darkOrange: #894b00;
  --pumpkinOrange: #c46c00;
  --lightGreen: #1bc47d;
  --green: #0ca52f;
  --darkGreen: #026c45;
  --slytherinGreen: #013220;
}
```

File: src/App/App.jsx (hunk 2 above)
Comment (Dev1LtJ) [reply src/App/App.jsx:77]: Moved to:
1) ComponentDidMount
2) ComponentDidUpdate

File: src/App/App.jsx (hunk 1 above)
Comment (Dev1LtJ) [reply src/App/App.jsx:86]: Moved to:
1) ComponentDidMount;
2) ComponentDidUpdate.

**Review by maximsan — APPROVED** [review]
Looks fine! Good job. Approved!

**Review by YauhenKoipish — APPROVED** [review]

**Pull request merged**